rdr_service/api/message_broker_cloud_tasks_api.py

```diff
@@ -5,53 +5,81 @@
-
         message_body = data.get('requestBody')
-        with dao.session() as session:
-            for key, value in message_body.items():
-                field_name = key
-
-                value_date_time = None
-                if isinstance(value, str):
-                    try:
-                        value_date_time = parse(value)
-                    except ValueError:
-                        pass
-
-                message_event_date = MessageBrokerEventData(
-                    messageRecordId=message_record_id,
-                    eventType=event_type,
-                    eventAuthoredTime=event_authored_time,
-                    participantId=participant_id,
-                    fieldName=field_name
-                )
-                if isinstance(value_date_time, datetime.datetime):
-                    message_event_date.valueDatetime = value
-                elif isinstance(value, bool):
-                    message_event_date.valueBool = value
-                elif isinstance(value, dict):
-                    message_event_date.valueJson = value
-                elif isinstance(value, str):
-                    message_event_date.valueString = value
-                elif isinstance(value, int):
-                    message_event_date.valueInteger = value
-                dao.insert_with_session(session, message_event_date)
+
+        for key, value in message_body.items():
+            field_name = key
+
+            value_date_time = None
+            if isinstance(value, str):
+                try:
+                    value_date_time = parse(value)
+                except ValueError:
+                    pass
+
+            message_event_date = MessageBrokerEventData(
+                messageRecordId=message_record_id,
+                eventType=self.event_type,
+                eventAuthoredTime=event_authored_time,
+                participantId=participant_id,
+                fieldName=field_name
+            )
+            if isinstance(value_date_time, datetime.datetime):
+                message_event_date.valueDatetime = value
+            elif isinstance(value, bool):
+                message_event_date.valueBool = value
+            elif isinstance(value, dict):
+                message_event_date.valueJson = value
+            elif isinstance(value, str):
+                message_event_date.valueString = value
+            elif isinstance(value, int):
+                message_event_date.valueInteger = value
+
+            inserted_obj = self.dao.insert(message_event_date)
+
+        self.call_informing_loop_task(inserted_obj)
 
         return '{"success": "true"}'
+
+    def call_informing_loop_task(self, obj):
+
+        if GAE_PROJECT != 'localhost' \
+                and 'informing_loop'.lower() \
+                in self.event_type:
+
+            payload = {}
+            informing_records = self.dao.get_informing_loop(
+                obj.messageRecordId,
+                self.event_type
```

Comment:
How do you know the records you get is informing records? It seem all events data will be send to you cloud task. Correct me if I miss anything.

Reply:
If `informing_loop` is in the event_type parsed from the payload. I can move lines 69-71 to after the loop for conditional test if that makes more sense?

Reply:
Never mind! I missed line 70...